ext/node/polyfills/process.ts

```diff
@@ -744,72 +888,57 @@ internals.__bootstrapNodeProcess = function (
-  internals.nodeProcessUnhandledRejectionCallback = (event) => {
-    if (process.listenerCount("unhandledRejection") === 0) {
-      // The Node.js default behavior is to raise an uncaught exception if
-      // an unhandled rejection occurs and there are no unhandledRejection
-      // listeners.
-      if (process.listenerCount("uncaughtException") === 0) {
-        throw event.reason;
-      }
-
-      event.preventDefault();
-      uncaughtExceptionHandler(event.reason, "unhandledRejection");
-      return;
-    }
-
-    event.preventDefault();
-    process.emit("unhandledRejection", event.reason, event.promise);
-  };
-
-  globalThis.addEventListener("error", (event) => {
-    if (process.listenerCount("uncaughtException") > 0) {
-      event.preventDefault();
-    }
-
-    uncaughtExceptionHandler(event.error, "uncaughtException");
-  });
-
-  globalThis.addEventListener("beforeunload", (e) => {
-    process.emit("beforeExit", process.exitCode || 0);
-    processTicksAndRejections();
-    if (core.eventLoopHasMoreWork()) {
-      e.preventDefault();
-    }
-  });
-
-  globalThis.addEventListener("unload", () => {
-    if (!process._exiting) {
-      process._exiting = true;
-      process.emit("exit", process.exitCode || 0);
-    }
-  });
-
-  // Initializes stdin
-  stdin = process.stdin = initStdin();
-
-  /** https://nodejs.org/api/process.html#process_process_stderr */
-  stderr = process.stderr = createWritableStdioStream(
-    io.stderr,
-    "stderr",
-  );
-
-  /** https://nodejs.org/api/process.html#process_process_stdout */
-  stdout = process.stdout = createWritableStdioStream(
-    io.stdout,
-    "stdout",
-  );
-
   process.setStartTime(Date.now());
   // @ts-ignore Remove setStartTime and #startTime is not modifiable
   delete process.setStartTime;
   delete internals.__bootstrapNodeProcess;
 };
 
+// deno-lint-ignore no-explicit-any
+let stderr_ = null as any;
+// deno-lint-ignore no-explicit-any
+let stdin_ = null as any;
+// deno-lint-ignore no-explicit-any
+let stdout_ = null as any;
+
+function getStdin(): Readable {
+  if (!stdin_) {
+    stdin_ = initStdin();
+    stdin = stdin_;
+    Object.defineProperty(process, "stdin", { get: () => stdin_ });
+  }
+  return stdin_;
+}
+
+/** https://nodejs.org/api/process.html#process_process_stdout */
+function getStdout(): Writable {
+  if (!stdout_) {
+    stdout_ = createWritableStdioStream(
+      io.stdout,
+      "stdout",
+    );
+    stdout = stdout_;
+    Object.defineProperty(process, "stdout", { get: () => stdout_ });
+  }
+  return stdout_;
+}
+
+/** https://nodejs.org/api/process.html#process_process_stderr */
+function getStderr(): Writable {
+  if (!stderr_) {
+    stderr_ = createWritableStdioStream(
+      io.stderr,
+      "stderr",
+    );
+    stderr = stderr_;
+    Object.defineProperty(process, "stderr", { get: () => stderr_ });
+  }
+  return stderr_;
+}
+
+core.setNextTickCallback(processTicksAndRejections);
+core.setMacrotaskCallback(runNextTicks);
```

Comment:
Ooops, I'm not sure if it's a good idea to do that during snapshotting. Could you move it to `nodeBootstrap`?

Reply:
Done